File: update_heartbeat.py
import json
import time

try:
    with open('/root/.openclaw/workspace/memory/heartbeat-state.json', 'r') as f:
        data = json.load(f)
except Exception:
    data = {"lastChecks": {}}

now = int(time.time())
data["lastChecks"]["youtube"] = now
data["lastChecks"]["signals"] = now

with open('/root/.openclaw/workspace/memory/heartbeat-state.json', 'w') as f:
    json.dump(data, f, indent=2)

File: update_heartbeat.js
const fs = require('fs');
const path = '/root/.openclaw/workspace/memory/heartbeat-state.json';
let state = { lastChecks: {} };
if (fs.existsSync(path)) {
  state = JSON.parse(fs.readFileSync(path, 'utf8'));
}
state.lastChecks.youtube = Math.floor(Date.now() / 1000);
fs.writeFileSync(path, JSON.stringify(state, null, 2));
